Sources/SecureXPC/Client/XPCClient.swift

```diff
@@ -157,87 +157,153 @@ public class XPCClient {
+    ///   - withResponse: A function or closure to receive the response.
     public func send<R: Decodable>(route: XPCRouteWithoutMessageWithReply<R>,
-                                   withReply reply: @escaping XPCReplyHandler<R>) throws {
-        let encoded = try Request(route: route.route).dictionary
-        try sendWithReply(encoded: encoded, withReply: reply)
+                                   withResponse handler: @escaping XPCResponseHandler<R>) {
+        do {
+            let encoded = try Request(route: route.route).dictionary
+            sendWithResponse(encoded: encoded, withResponse: handler)
+        } catch {
+            handler(.failure(.encodingError(String(describing: error))))
+        }
     }
     
-    /// Sends a message and provides the reply as either a message on success or an error on failure.
+    /// Sends a message and provides the response as either a reply on success or an error on failure.
     ///
     /// - Parameters:
     ///    - message: Message to be sent.
     ///    - route: The server route which should handle this message.
-    ///    - withReply: A function or closure to receive the message's reply.
-    /// - Throws: If unable to encode the message or route or the server throws an error. No error will be thrown if communication with the server fails.
+    ///    - withResponse: A function or closure to receive the message's response.
     public func sendMessage<M: Encodable, R: Decodable>(_ message: M,
                                                         route: XPCRouteWithMessageWithReply<M, R>,
-                                                        withReply reply: @escaping XPCReplyHandler<R>) throws {
-        let encoded = try Request(route: route.route, payload: message).dictionary
-        try sendWithReply(encoded: encoded, withReply: reply)
+                                                        withResponse handler: @escaping XPCResponseHandler<R>) {
+        do {
+            let encoded = try Request(route: route.route, payload: message).dictionary
+            sendWithResponse(encoded: encoded, withResponse: handler)
+        } catch {
+            handler(.failure(.encodingError(String(describing: error))))
+        }
     }
     
-    /// Does the actual work of sending an XPC message which receives a reply.
-    private func sendWithReply<R: Decodable>(encoded: xpc_object_t,
-                                             withReply reply: @escaping XPCReplyHandler<R>) throws {
-        xpc_connection_send_message_with_reply(try getConnection(), encoded, nil, { xpcResponse in
+    /// Does the actual work of sending an XPC message which receives a response.
+    private func sendWithResponse<R: Decodable>(encoded: xpc_object_t,
+                                                withResponse handler: @escaping XPCResponseHandler<R>) {
+        // Get the connection or inform the handler of failure and return
+        var connection: xpc_connection_t?
```

Comment:
I don't think Swift requires this to be mutable or optional, because it's always initialized once in all code paths

Reply:
But it's only initialized in one code path? If `try getConnection()` succeeds it's initialized, but if the `catch` block is entered then it's not initialized. Maybe I'm missing something.

Reply:
The `catch` path should `return`, then what I said holds true.

Minimal sample:

```swift
func thrower() throws -> Int { 123 }

func demo() {
	let i: Int
	
	do {
		i = try thrower()
	} catch {
		return
	}
	
	print(i)
}

demo()
```

Reply:
Ah yup, that makes complete sense. Will do.